docs/rule/no-bare-strings.md

```diff
@@ -22,32 +22,20 @@ This rule **allows** the following:
  The following values are valid configuration:
 
 * boolean -- `true` for enabled / `false` for disabled
-* array -- an array of allowlisted strings
+* array -- An array of allowlisted strings (extends the default config)
 * object -- An object with the following keys:
-  * `allowlist` -- An array of allowlisted strings
+  * `allowlist` -- An array of allowlisted strings (extends the default config)
   * `globalAttributes` -- An array of attributes to check on every element.
   * `elementAttributes` -- An object whose keys are tag names and value is an array of attributes to check for that tag name.
```

Comment:
@dcyriller I want to make sure that all of these options extend the default config now instead of the previous replace behavior. Can you make sure this behavior is documented and tested for `globalAttributes` and `elementAttributes`?

```suggestion
  * `globalAttributes` -- An array of attributes to check on every element (extends the default config)
  * `elementAttributes` -- An object whose keys are tag names and value is an array of attributes to check for that tag name (extends the default config)
```

My top priority for this PR is changing the default behavior of these existing options to all be consistent like this.

I would actually be fine if we don't add any new options in this PR (`includeDefaultAllowList`, `ignoreDefaultGlobalAttributes`, `ignoreDefaultElementAttributes`). These options can be added at a later time if someone requests them (I'm not sure anyone actually wants to ignore the default config). Maybe we should just leave them out to keep the rule simpler.